keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py

```diff
@@ -0,0 +1,385 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DarkNet models for KerasCV. """
+import copy
+
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.csp_darknet.csp_darknet_backbone_presets import (
+    backbone_presets,
+)
+from keras_cv.models.backbones.csp_darknet.csp_darknet_backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.models.backbones.csp_darknet.darknet_utils import (
+    CrossStagePartial,
```

Comment:
FWIW, these blocks are still all wrappers around `keras.Layer` instead of `apply_xyz` functions.

I think at some point we skipped porting these to the `apply_xyz` approach to avoid breaking our pre-trained weights, but I'm not sure if this is really important at all.

Probably not a TODO for this PR, but something we should write a TODO for in the codebase.